### PR TITLE
Fix VDF prover failures on Windows

### PR DESCRIPTION
**Pull request by wjblanke**
The root issue is that labs() was being used instead of llabs(). labs() is undefined (on windows only apparently) if the value tested is too large. This was causing the Pullmark reducer to misbehave and reduction not to happen correctly in rare cases. That resulted in too large values and sometimes buffer overflows causing heap corruption in some cases. In the process fixed some other things the compiler was not happy about but doesn't really affect the behavior.

### CODE REVIEW

File: src/Reducer.h [hunk 1]
```diff
@@ -229,7 +229,8 @@ bool bLZCHasHW=false;
 
       // The condition (abs(v_) | abs(x_)) <= THRESH protects against
       // overflow
-      below_threshold = (labs(v_) | labs(x_)) <= THRESH ? 1 : 0;
+      // must use llabs() as labs() is undefined for too large numbers
+      below_threshold = (llabs(v_) | llabs(x_)) <= THRESH ? 1 : 0;
```
Comment (arvidn) [inline src/Reducer.h:233]: technically, `llabs()` is supposed to be declared in `<cstdlib>`. https://en.cppreference.com/w/cpp/numeric/math/abs
Perhaps it ends up being included indirectly

File: src/verifier_test.cpp [hunk 2]
```diff
@@ -22,13 +24,45 @@ std::vector<uint8_t> HexToBytes(const char *hex_proof) {
...
+    // Test overflow for slow prover (part of challenge b'\xa6\xc4%X\x17O\xb1\xee\xdcd')
+    int l,k;
+    ApproximateParameters(90909, l, k);
+
+    cout << "ApproximateParameters for 90909 l: " << l << " k: " << k << endl;
+
+    std::vector<uint8_t> challenge_hash=HexToBytes("a6c42558174fb1eedc64");
+    std::vector<uint8_t> x_s=HexToBytes("0300aca4849458af5c557710c80f21519f196907764d2d55c9b70581a90d49ca7b3201ad6a9da836429e6592c200e965434f0100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000");
+
+    ByteArray ba=prove_wrapper(challenge_hash.data(), challenge_hash.size(), x_s.data(), x_s.size(), 512, 90909);
```
Comment (arvidn) [inline src/verifier_test.cpp:55]: `prove_wrapper()` returns a heap allocation the caller is expected to free. You're leaking memory here as far as I can tell

**Review by arvidn — DISMISSED** [review]
the memory leak is in a test, so not a big deal probably.
It might deceive a future reader of the code though

File: src/verifier_test.cpp (hunk 2 above)
Comment (wjblanke) [reply src/verifier_test.cpp:55]: yup! I thought it was a class but its C. its just a test but freed now

File: src/Reducer.h (hunk 1 above)
Comment (wjblanke) [reply src/Reducer.h:233]: ya unsure, the original code was abs() and depended on the overloads. maybe the compiler wasn't happy with that for the hwvdf changes, but that got switched to the wrong explicit form.

**Review by arvidn — APPROVED** [review]
*technically* it should be `delete[] ba.data;`, but I was also under the impression that we had a function to free a `ByteArray` object. mostly just to ensure we use the correct heap in case this is exported over a DLL boundary.
A minor issue though.

**Review by arvidn — APPROVED** [review]

**Pull request merged**